### PR TITLE
Testing with Mocks: Sms Validator

### PR DESCRIPTION
**Pull request by denisdimitrov**

### CODE REVIEW

File: src/smsvalidator/Gateway.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public class Gateway {
+    public final Validator validator;
+    public final Sender sender;
+
+    public Gateway(Validator validator, Sender sender) {
+        this.validator = validator;
+        this.sender = sender;
+    }
+
+    public boolean check(Sms sms) {
```
Comment (EmilGeorgiev) [inline src/smsvalidator/Gateway.java]: send(SMS sms)

File: src/smsvalidator/Sender.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package smsvalidator;
+
+/**
+ * Created by clouway on 15.08.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public interface Sender {
+    boolean send(Sms sms);
```
Comment (EmilGeorgiev) [inline src/smsvalidator/Sender.java]: add documentation of classes, interfaces and methods

**Comment by denisdimitrov** [comment]
Fixed everything that you asked for.

**Comment by EmilGeorgiev** [comment]
LGTM

**Pull request merged**